editor/gui/editor_scroll_box.cpp

```diff
@@ -0,0 +1,298 @@
+	} else if (Input::get_singleton()->is_key_pressed(Key::SHIFT)) {
+		scroll_bar->set_value(scroll_bar->get_value() + scroll_bar->get_page() * (p_forward ? 1 : -1));
+	} else {
+		scroll_bar->set_value(scroll_bar->get_value() + (scroll_bar->get_page() * 0.5) * (p_forward ? 1 : -1));
+	}
+}
+
+// Update the buttons visibility and disabled state.
+void EditorScrollBox::_update_buttons() {
+	bool show_arrows = control && control->get_size()[is_vertical() ? 1 : 0] > scroll_container->get_size()[is_vertical() ? 1 : 0];
+	first_button->set_visible(show_arrows);
+	second_button->set_visible(show_arrows);
+
+	if (show_arrows) {
+		_update_disabled_buttons();
+	}
+}
+
+void EditorScrollBox::_update_buttons_icon_and_tooltip() {
+	if (!is_vertical()) {
+		if (is_layout_rtl()) {
+			second_button->set_button_icon(theme_cache.arrow_left);
+			second_button->set_accessibility_name(TTRC("Scroll Left"));
+			second_button->set_tooltip_text(TTRC("Scroll Left\nHold Ctrl to scroll to the begin.\nHold Shift to scroll one page."));
+			first_button->set_button_icon(theme_cache.arrow_right);
+			first_button->set_accessibility_name(TTRC("Scroll Right"));
+			first_button->set_tooltip_text(TTRC("Scroll Right\nHold Ctrl to scroll to the end.\nHold Shift to scroll one page."));
+		} else {
+			first_button->set_button_icon(theme_cache.arrow_left);
+			first_button->set_accessibility_name(TTRC("Scroll Left"));
+			first_button->set_tooltip_text(TTRC("Scroll Left\nHold Ctrl to scroll to the begin.\nHold Shift to scroll one page."));
+			second_button->set_button_icon(theme_cache.arrow_right);
+			second_button->set_accessibility_name(TTRC("Scroll Right"));
+			second_button->set_tooltip_text(TTRC("Scroll Right\nHold Ctrl to scroll to the end.\nHold Shift to scroll one page."));
+		}
+	} else {
+		first_button->set_button_icon(theme_cache.arrow_up);
+		first_button->set_accessibility_name(TTRC("Scroll Up"));
+		first_button->set_tooltip_text(TTRC("Scroll Up\nHold Ctrl to scroll to the begin.\nHold Shift to scroll one page."));
+		second_button->set_button_icon(theme_cache.arrow_down);
+		second_button->set_accessibility_name(TTRC("Scroll Down"));
+		second_button->set_tooltip_text(TTRC("Scroll Down\nHold Ctrl to scroll to the end.\nHold Shift to scroll one page."));
+	}
+}
+
+void EditorScrollBox::_update_disabled_buttons() {
+	ScrollBar *scroll_bar = is_vertical() ? (ScrollBar *)scroll_container->get_v_scroll_bar() : (ScrollBar *)scroll_container->get_h_scroll_bar();
+	if (!is_vertical() && is_layout_rtl()) {
+		first_button->set_disabled(scroll_bar->get_value() + scroll_bar->get_page() == scroll_bar->get_max());
+		second_button->set_disabled(scroll_bar->get_value() == 0);
+	} else {
+		first_button->set_disabled(scroll_bar->get_value() == 0);
+		second_button->set_disabled(scroll_bar->get_value() + scroll_bar->get_page() == scroll_bar->get_max());
+	}
+}
+
+bool EditorScrollBox::_set(const StringName &p_name, const Variant &p_value) {
+	if (p_name == "vertical") {
```

Comment:
This looks like unused code. The property is not defined.
Also makes `_validate_property()` not called.

Reply:
> This looks like unused code. The property is not defined. Also makes `_validate_property()` not called.

It extends `BoxContainer` that's why i have this method to update the layout when `BoxContainer::vertical` is changed.